Core/Src/main.c

```diff
@@ -87,14 +89,38 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();       // SPI is initialized here
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  uint8_t inputData[3] = {0x01, 0x80, 0x00};
+  uint8_t outputData[3] = {0};
+  uint16_t dataSize = 3;
+  uint32_t timeout = 10; // ms
+
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET); // Start with CS pin set or it may not work apparently
```

Comment:
Since the CS line is idle HIGH, we want to explicitly set it HIGHhere like you did. Depending on if there is a pull down configured, the default could be LOW. Or, if there is no configuration on a pull up or down, then it could be floating (undefined behavior). Always best to be explicit to make sure :)